### PR TITLE
Simplified implementation of _sre.ascii_iscased().

### PR DESCRIPTION
**Pull request by sir-sigurd**

### CODE REVIEW

**Review by zhangyangyu — APPROVED** [review]

File: Modules/_sre.c [hunk 1]
```diff
@@ -293,7 +288,7 @@ _sre_ascii_iscased_impl(PyObject *module, int character)
 /*[clinic end generated code: output=4f454b630fbd19a2 input=9f0bd952812c7ed3]*/
 {
     unsigned int ch = (unsigned int)character;
-    return ch != sre_lower_ascii(ch) || ch != sre_upper_ascii(ch);
+    return ch <= 'z' && Py_ISALPHA(ch);
```
Comment (serhiy-storchaka) [inline Modules/_sre.c]: I would use `127` instead of `'z'`.

**Review by benjaminp — CHANGES_REQUESTED** [review]
Agreed that `< 128` would be better.

**Comment by serhiy-storchaka** [comment]
Performance is not critical here, but readability matters. It took a while for me to figure out why this code is written in such way and to ensure that it is correct. It can be looking more cryptic for other readers. The meaning of `< 128` is more clear: it is for ASCII only.

**Comment by sir-sigurd** [comment]
I have made the requested changes; please review again.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**